model/riscv_pmp_regs.sail

```diff
@@ -102,59 +102,52 @@ bitfield Pmpcfg_ent : bits(8) = {
   R : 0         /* read */
 }
 
-register pmp0cfg  : Pmpcfg_ent
-register pmp1cfg  : Pmpcfg_ent
-register pmp2cfg  : Pmpcfg_ent
-register pmp3cfg  : Pmpcfg_ent
-register pmp4cfg  : Pmpcfg_ent
-register pmp5cfg  : Pmpcfg_ent
-register pmp6cfg  : Pmpcfg_ent
-register pmp7cfg  : Pmpcfg_ent
-register pmp8cfg  : Pmpcfg_ent
-register pmp9cfg  : Pmpcfg_ent
-register pmp10cfg : Pmpcfg_ent
-register pmp11cfg : Pmpcfg_ent
-register pmp12cfg : Pmpcfg_ent
-register pmp13cfg : Pmpcfg_ent
-register pmp14cfg : Pmpcfg_ent
-register pmp15cfg : Pmpcfg_ent
-
-/* PMP address configuration */
-
-register pmpaddr0  : xlenbits
-register pmpaddr1  : xlenbits
-register pmpaddr2  : xlenbits
-register pmpaddr3  : xlenbits
-register pmpaddr4  : xlenbits
-register pmpaddr5  : xlenbits
-register pmpaddr6  : xlenbits
-register pmpaddr7  : xlenbits
-register pmpaddr8  : xlenbits
-register pmpaddr9  : xlenbits
-register pmpaddr10 : xlenbits
-register pmpaddr11 : xlenbits
-register pmpaddr12 : xlenbits
-register pmpaddr13 : xlenbits
-register pmpaddr14 : xlenbits
-register pmpaddr15 : xlenbits
+register pmpcfg_n : vector(64, dec, Pmpcfg_ent)
+register pmpaddr_n : vector(64, dec, xlenbits)
 
 /* Packing and unpacking pmpcfg regs for xlen-width accesses */
 
-val pmpReadCfgReg : forall 'n, 0 <= 'n < 4 . (atom('n)) -> xlenbits
-function pmpReadCfgReg(n) = {
-  if   sizeof(xlen) == 32
-  then match n {
-         0 => append(pmp3cfg.bits(),  append(pmp2cfg.bits(),  append(pmp1cfg.bits(),  pmp0cfg.bits()))),
-         1 => append(pmp7cfg.bits(),  append(pmp6cfg.bits(),  append(pmp5cfg.bits(),  pmp4cfg.bits()))),
-         2 => append(pmp11cfg.bits(), append(pmp10cfg.bits(), append(pmp9cfg.bits(),  pmp8cfg.bits()))),
-         3 => append(pmp15cfg.bits(), append(pmp14cfg.bits(), append(pmp13cfg.bits(), pmp12cfg.bits()))),
-         _ => internal_error(__FILE__, __LINE__, "Unexpected pmp config reg read")
-       }
-  else match n { //  sizeof(xlen) >= 64
-         0 => append(pmp7cfg.bits(),  append(pmp6cfg.bits(),  append(pmp5cfg.bits(),  append(pmp4cfg.bits(),  append(pmp3cfg.bits(),  append(pmp2cfg.bits(),  append(pmp1cfg.bits(),  pmp0cfg.bits()))))))),
-         2 => append(pmp15cfg.bits(), append(pmp14cfg.bits(), append(pmp13cfg.bits(), append(pmp12cfg.bits(), append(pmp11cfg.bits(), append(pmp10cfg.bits(), append(pmp9cfg.bits(),  pmp8cfg.bits()))))))),
-         _ => internal_error(__FILE__, __LINE__, "Unexpected pmp config reg read")
-       }
+function pmpReadCfgReg(n : range(0, 15)) -> xlenbits = {
+  if sizeof(xlen) == 32
+  then {
+    pmpcfg_n[n*4 + 3].bits() @
+    pmpcfg_n[n*4 + 2].bits() @
+    pmpcfg_n[n*4 + 1].bits() @
+    pmpcfg_n[n*4 + 0].bits()
+  }
+  else {
+    assert(n % 2 == 0, "Unexpected pmp config reg read");
```

Comment:
This should cause an illegal instruction trap, not an error message that stops the simulation and I can't tell how of if this responds.